File: backend/furniture/src/main/java/com/khumaloterrance/furniture/dto/Purchase.java
package com.khumaloterrance.furniture.dto;

import com.khumaloterrance.furniture.entity.Address;
import com.khumaloterrance.furniture.entity.Customer;
import com.khumaloterrance.furniture.entity.ItemToPurchase;
import com.khumaloterrance.furniture.entity.Order;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Order order;
    private Set<ItemToPurchase> itemsToPurchase;
}

File: frontend/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { ItemToPurchase } from "./item-to-purchase";
import { Order } from "./order";

export class Purchase {
    customer: Customer | undefined;
    shippingAddress: Address | undefined;
    order: Order | undefined;
    itemsToPurchase: ItemToPurchase[] | undefined;
  orderTrackingNumber: any;
}
